js/src/bitswap.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+const statsTests = require('./utils/stats')
+chai.use(dirtyChai)
+const CID = require('cids')
+
+module.exports = (common) => {
+  describe('.bitswap online', () => {
+    let ipfs
+    let withGo
+    const key = 'QmUBdnXXPyoDFXj3Hj39dNJ5VkN3QFRskXxcGaYFBB8CNR'
+
+    before(function (done) {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(60 * 250)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist()
+          ipfs = node
+          ipfs.block.get(key)
+            .then(() => {})
+            .catch(() => {})
+          ipfs.id((err, id) => {
+            expect(err).to.not.exist()
+            withGo = id.agentVersion.startsWith('go-ipfs')
+            setTimeout(done, 250) //Give block.get time to take effect
+          })
```

Comment:
We now have a `spawnNodeWithId` to help reduce this code duplication around the project.

https://github.com/ipfs/interface-ipfs-core/blob/master/js/src/utils/spawn.js#L5,L14

nit: I'm not a huge fan of the timeout here. Does it really take a lot of time for the key to enter the want list?

Reply:
No, the timeout appeared to be cruft leftover from debugging other errors in the wantlist return that had to do with the disparity between go and the js implementations. It works fine without for me in both js-ipfs and js-ipfs-api.

Switched to spawnWithNodeId too.